src/network/networkd-link.c

```diff
@@ -617,6 +618,11 @@ static int link_new(Manager *manager, sd_netlink_message *message, Link **ret) {
         if (r < 0)
                 log_link_debug_errno(link, r, "MAC address not found for new device, continuing without");
 
+        _cleanup_close_ int fd = -1;
+        r = ethtool_get_permanent_macaddr(&fd, link->ifname, &link->permanent_mac);
```

Comment:
hmm, btw, what happens on devices that have no permanent MAC addresses? i.e. those embedded ethernet devices that have no ROM? Will this call fail? or does this return 00:00:00:00:00:00?

Reply:
Hmm, I do not know... I do not have such a device.

Reply:
Hello @yuwata , really appreciate your work, thanks a lot for you PR.

Non-physical interfaces like bridges or veth do not have permanent MAC and it is returned as all-zeroes:

```
# ethtool -P br0
Permanent address: 00:00:00:00:00:00
# ethtool -P veth1748465eth0
Permanent address: 00:00:00:00:00:00
```

Reply:
@serge-name Thank you for the comment!